deeplake/util/version_control.py

```diff
@@ -281,6 +288,129 @@ def checkout(
         ) from e
 
 
+def delete_branch(
```

Comment:
Can we approach this from OOP perspective? Create a class and bunch of methods inside? Then call few of them here, some parts of the code could be reused in the future and instead of rewriting it again and again, it will be great to have a modular code

Reply:
I think we will want to redo it as more OOP in the coming weeks as we look at some refactoring around the overall version-control tracking in on disk. 

For this specific feature change, it seems best to keep with the current style and then do the OO change as a separate step to keep it simpler?